Limber/MIXMockServer.cpp

```diff
@@ -0,0 +1,274 @@
+                        rosterUpdatePayload->addItem(itemPayload);
+                        session->sendElement(IQ::createRequest(IQ::Set, iq->getFrom(), iq->getID(), rosterUpdatePayload));
+
+                    } else {
+                        SWIFT_LOG(debug) << "Initial roster not requested by client." <<std::endl;
+                    }
+                } else {
+                    SWIFT_LOG(debug) << "Channel " << *channelJID << "not hosted on domain." <<std::endl;
+                }
+            } else {
+                session->sendElement(IQ::createError(iq->getFrom(), iq->getID(), ErrorPayload::BadRequest, ErrorPayload::Cancel));
+            }
+        }
+
+        void handleLeave(IQ::ref iq, std::shared_ptr<ServerFromClientSession> session, MIXLeave::ref incomingLeavePayload) {
+            if (auto channelJID = incomingLeavePayload->getChannel()) {
+                SWIFT_LOG(debug) << "Request: Channel Leave - " << *channelJID << std::endl;
+                if (mixChannelRegistry_->hasMIXChannel(*channelJID)) {
+
+                    session->sendElement(IQ::createResult(iq->getFrom(), iq->getTo(), iq->getID(), std::make_shared<MIXLeave>()));
+
+                    auto i = rosterMap_.find(iq->getFrom().toBare());
+                    if (i != rosterMap_.end()) {
+                        auto roster = i->second;
+                        roster->removeContact(*channelJID);
+
+                        auto rosterUpdatePayload = std::make_shared<RosterPayload>();
+                        auto itemPayload = RosterItemPayload(*channelJID, channelJID->getNode(), RosterItemPayload::Subscription::Remove);
+                        itemPayload.setMIXChannel(true);
+                        rosterUpdatePayload->addItem(itemPayload);
+
+                        session->sendElement(IQ::createRequest(IQ::Set, iq->getFrom(), iq->getID(), rosterUpdatePayload));
+                    } else {
+                        SWIFT_LOG(debug) << "Initial roster not requested by client." <<std::endl;
+                    }
+                } else {
+                    SWIFT_LOG(debug) << "Channel " << *channelJID << "not hosted on domain." <<std::endl;
+                }
+            } else {
+                session->sendElement(IQ::createError(iq->getFrom(), iq->getID(), ErrorPayload::BadRequest, ErrorPayload::Cancel));
+            }
+        }
+
+        MIXJoin::ref createJoinResult(const std::unordered_set<std::string>& nodes) {
+            auto joinResultPayload = std::make_shared<MIXJoin>();
+            for (auto node : nodes) {
+                joinResultPayload->addSubscription(node);
+            }
+            joinResultPayload->setJID(JID("123456#coven@example.com"));
+            return joinResultPayload;
+        }
+
+        std::shared_ptr<RosterPayload> getRosterPayloadFromXMPPRoster(std::shared_ptr<XMPPRosterImpl> roster) {
+            auto rosterPayload = std::make_shared<RosterPayload>();
```

Comment:
`getFoo()` usually describes a getter method. This seems to be a) not a method to retrieve a member and b) a method that does not otherwise access members, so can be static, right?